### PR TITLE
Backport workflow improvements and increment version

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
* Backport #439
* Increment version to 2.5.0 

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/publish-snapshots.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Publish snapshots to maven
+
+on:
+  push:
+    branches:
+      - main
```
Comment (reta) [inline .github/workflows/publish-snapshots.yml:6]: Probably `main` is not needed here, since the pull request goes to `2.x` branch

Comment (VachaShah) [reply .github/workflows/publish-snapshots.yml:6]: Totally!

**Review by reta — APPROVED** [review]

**Comment by VachaShah** [comment]
@reta To publish these snapshots though, I would need to increment version in this branch so the snapshot can be of the next version. Let me add it to this PR itself.

File: gradle.properties [hunk 2]
```diff
@@ -1 +1 @@
-systemProp.version = 2.4.0
+systemProp.version = 2.5.0
```
Comment (reta) [inline gradle.properties:1]: 👍

**Review by reta — APPROVED** [review]

**Comment by VachaShah** [comment]
@gaiksaya Mentioned that the publish snapshots workflow might not need to be backported to 2.x, we can just add 2.x to the branch list on main. Doing that in PR #451

**Comment by VachaShah** [comment]
Will merge #451 first, so that when this PR is merged, the snapshot can be published for 2.5.0.

**Pull request merged**